### PR TITLE
#487 Update Checkstyle to 6.12.1

### PR DESCRIPTION
**Pull request by mkordas**

### CODE REVIEW

File: qulice-maven-plugin/src/it/pmd-violations/src/main/java/com/qulice/foo/Violations.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public static void print() {
          * Constructor.
          * @param nam Name.
          */
-        public Foo(final String nam) {
+        Foo(final String nam) {
```
Comment (mkordas) [inline qulice-maven-plugin/src/it/pmd-violations/src/main/java/com/qulice/foo/Violations.java]: Change needed as of checkstyle/checkstyle#1537

File: qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml [hunk 2]
```diff
@@ -288,6 +288,7 @@
             <property name="tokens" value="CTOR_DEF,METHOD_DEF,FOR_EACH_CLAUSE,LITERAL_CATCH"/>
         </module>
         <module name="Indentation"/>
+        <module name="CommentsIndentation"/>
```
Comment (mkordas) [inline qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml]: New check: checkstyle/checkstyle#333

File: qulice-checkstyle/src/main/java/com/qulice/checkstyle/CheckstyleValidator.java [hunk 3]
```diff
@@ -112,10 +112,17 @@ public void validate(final Environment env) throws ValidationException {
...
+                new File(env.tempdir(), "checkstyle/checkstyle.cache");
+        if (!cacheFile.getParentFile().mkdirs()) {
+            throw new IllegalStateException("Unable to crate directories"
+                    + " needed for " + cacheFile.getPath()
+            );
+        }
         final Properties props = new Properties();
         props.setProperty(
             "cache.file",
-            new File(env.tempdir(), "checkstyle/checkstyle.cache").getPath()
```
Comment (mkordas) [inline qulice-checkstyle/src/main/java/com/qulice/checkstyle/CheckstyleValidator.java]: Checkstyle has never been able to create directory structure for cache file, but previously exception was swallowed: https://github.com/checkstyle/checkstyle/commit/ea9ff3965d9c657b5b7bd2d969e075f430635c47#diff-08bcb3012a3f44d264098cc63eabb365L153

File: qulice-checkstyle/src/main/java/com/qulice/checkstyle/CheckstyleValidator.java [hunk 4]
```diff
@@ -73,11 +73,7 @@ public void validate(final Environment env) throws ValidationException {
             return;
         }
         final Checker checker;
-        try {
-            checker = new Checker();
-        } catch (final CheckstyleException ex) {
-            throw new IllegalStateException("Failed to create checker", ex);
-        }
+        checker = new Checker();
```
Comment (mkordas) [inline qulice-checkstyle/src/main/java/com/qulice/checkstyle/CheckstyleValidator.java]: API has slightly changed

**Comment by mkordas** [comment]
I'm happy to answer any questions you may have to this PR :)

**Comment by krzyk** [comment]
@yegor256 can we assign someone to this PR?

**Comment by davvd** [comment]
@mkordas Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@yegor256 please review, thanks

**Comment by krzyk** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@krzyk I'm curious, why all checks are passed (in Travis), but there string concatenation inside... what do you think?

**Comment by krzyk** [comment]
@yegor256 strange, same problem with brackets

**Comment by krzyk** [comment]
@yegor256 this is a specific case, when the last elemen in concatenation is not a string, in such case there will be no error

**Comment by krzyk** [comment]
@yegor256 reported #502

**Comment by yegor256** [comment]
@mkordas please, remove string concatenation, we're against it in general, see http://www.yegor256.com/2014/06/19/avoid-string-concatenation.html

**Comment by mkordas** [comment]
@yegor256 done

**Comment by krzyk** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@yegor256 Thanks so much! Your account was topped up for **18 mins** (transaction ID is `000-843f9470`, the task took 407 hours and 13 mins). review comments (c=3) added as a bonus. +18 added to your rating, at the moment it is: [+9428](http://www.netbout.com/b/35008?open=rating)

**Comment by davvd** [comment]
@rultor deploy pls

**Pull request merged**